test/unit/client.test.js

```diff
@@ -1037,27 +1037,13 @@ test('Content length too big (string)', (t) => {
 
 test('Content length exceeds max heap limit', (t) => {
   t.plan(4);
-  const percentage = 0.8;
-  const HEAP_SIZE_LIMIT_WITH_BUFFER = Number(
-    require('v8').getHeapStatistics().heap_size_limit * percentage
-  );
-  const contentLength = buffer.constants.MAX_STRING_LENGTH - 1;
+  const percentage = 0.01;
+  const HEAP_SIZE_LIMIT = require('v8').getHeapStatistics().heap_size_limit;
+  const contentLength = Math.round(HEAP_SIZE_LIMIT * percentage + 1);
   const memoryCircuitBreaker = {
     enabled: true,
     maxPercentage: percentage,
   };
-  // Simulate allocation of bytes
-  const memoryAllocations = [];
-  while (process.memoryUsage().heapUsed + contentLength <= HEAP_SIZE_LIMIT_WITH_BUFFER) {
-    const allocation = 50 * 1024 * 1024; // 50MB
-    const numbers = allocation / 8;
-    const arr = [];
-    arr.length = numbers;
-    for (let i = 0; i < numbers; i++) {
-      arr[i] = i;
-    }
-    memoryAllocations.push(arr);
-  }
```

Comment:
Remove this block of code because `memoryAllocations` is not being used anywhere.